### PR TITLE
conferences: custom start_date range filter

### PR DESCRIPTION
**Pull request by harunurhan**
* start_date range filter that handles special values like `upcoming`
and `all`

INSPIR-2969

### CODE REVIEW

File: backend/tests/integration/records/views/test_views_conferences.py [hunk 1]
```diff
@@ -264,3 +266,63 @@ def test_date_range_contains_conferences_filter(api_client, db, es, create_recor
     assert 1 in found_recids
     assert 2 in found_recids
     assert 3 in found_recids
+
+
+@freeze_time(datetime(2019, 9, 15, 0, 0, 0))  # today is the 9th of September, 2019
```
Comment (drjova) [inline backend/tests/integration/records/views/test_views_conferences.py]:
```suggestion
@freeze_time('2019-09-15')  # today is the 9th of September, 2019
```

Comment (drjova) [reply backend/tests/integration/records/views/test_views_conferences.py]: much easier to read :)

File: backend/tests/integration/records/views/test_views_conferences.py [hunk 2]
```diff
@@ -264,3 +266,63 @@ def test_date_range_contains_conferences_filter(api_client, db, es, create_recor
...
+def test_conferences_start_date_range_filter(api_client, db, es, create_record):
```
Comment (drjova) [inline backend/tests/integration/records/views/test_views_conferences.py]: I'd separate the tests for each case, cleaner and easier to understand. wdyt?

File: backend/inspirehep/search/facets.py [hunk 3]
```diff
@@ -52,6 +55,24 @@ def inner(values):
...
 
+def conferences_start_date_range_filter():
+    date_range_filter = range_filter("opening_date")
+
+    def inner(values):
+        value = values and values[0]
+
+        if value == "upcoming":
+            today = datetime.today().strftime("%Y-%m-%d")
+            return Range(**{"opening_date": {"gte": today}})
```
Comment (drjova) [inline backend/inspirehep/search/facets.py]: we can use something more elegant https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html#ranges-on-dates

File: backend/inspirehep/records/config.py [hunk 4]
```diff
@@ -489,7 +490,7 @@
     "records-conferences": {
         "filters": {
             "subject": must_match_all_filter("inspire_categories.term"),
-            "start_date": range_filter("opening_date"),
+            "start_date": conferences_start_date_range_filter(),
             "contains": date_range_contains_conferences_filter(),
```
[the remark this answers is not in the document]
Comment (harunurhan) [reply backend/inspirehep/records/config.py]: yeah I am not happy with naming but `conferences_date_range_contains`, contains what? :smile: and `conference_date_range_contains_conferences` looks weird :P

File: backend/tests/integration/records/views/test_views_conferences.py (hunk 1 above)
Comment (harunurhan) [reply backend/tests/integration/records/views/test_views_conferences.py]: was just following the codebase styles 😄

File: backend/tests/integration/records/views/test_views_conferences.py (hunk 2 above)
Comment (harunurhan) [reply backend/tests/integration/records/views/test_views_conferences.py]: I was thinking about this but there is repetitive part for whole cases (creating 3-4 conferences) , which looked annoying little bit, but I can separate.

Probably pytest has better way to handle this, like test clases with `setup` method, but didn't wanna introduce it.

File: backend/inspirehep/records/config.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (drjova) [reply backend/inspirehep/records/config.py]: still is a bit better imo or ``conferences_date_range_contains_other_conferences`` smth like that

File: backend/inspirehep/search/facets.py (hunk 3 above)
Comment (harunurhan) [reply backend/inspirehep/search/facets.py]: this makes it hard to test :/

**Pull request merged**